connection_pool.go

```diff
@@ -238,3 +247,72 @@ func (pool *ConnectionPool) createConnection() (*connection, error) {
 	// TODO: update workload
 	return newConn, nil
 }
+
+// startCleaner starts connectionCleaner if idleTime > 0.
+func (pool *ConnectionPool) startCleaner() {
+	if pool.conf.IdleTime > 0 && pool.cleanerChan == nil {
+		pool.cleanerChan = make(chan struct{}, 1)
+		go pool.connectionCleaner()
+	}
+}
+
+func (pool *ConnectionPool) connectionCleaner() {
+	const minInterval = time.Second
+
+	d := pool.conf.IdleTime
+
+	if d < minInterval {
+		d = minInterval
+	}
+	t := time.NewTimer(d)
+
```

Comment:
We don't need to check the idle status that often. Setting the checking interval to 60 seconds seems more appropriate to me.